# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/render_task_graph.rs
# Commit: c20cf2fa829a
# Full Hash: c20cf2fa829a19cb6b60909d05abcda40e77dd70
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-04-20 09:42:47
# Regressor Bug: 1764005
# File Overlap Count: 1
# Description:
#   Bug 1764005 - Add backdrop-filter primitive support r=gfx-reviewers,lsalzman,nical
#   
#   Add support for backdrop-filter primitive in WR. Each backdrop
#   filter primitive establishes a render task sub-graph. The primitive
#   instance registers itself as a resolve source to receive the
# ==============================================================================

diff -r d5b01209cf78 -r c20cf2fa829a gfx/wr/webrender/src/render_task_graph.rs
--- a/gfx/wr/webrender/src/render_task_graph.rs	Wed Apr 20 01:08:52 2022 +0000
+++ b/gfx/wr/webrender/src/render_task_graph.rs	Wed Apr 20 01:09:09 2022 +0000
@@ -785,7 +785,10 @@
                 parent_task.free_after = PassId::INVALID;
 
                 let child_task = &mut graph.tasks[child_id.index as usize];
-                child_task.free_after = child_task.free_after.min(render_on);
+
+                if child_task.free_after != PassId::INVALID {
+                    child_task.free_after = child_task.free_after.min(render_on);
+                }
             }
         }
 